internal/storage/smartcontract/smartcontract.go

```diff
@@ -87,9 +137,32 @@ func (s *Storage) ListSmartContracts(sort string, limit int64, offset int64) ([]
 	return smartcontracts, nil
 }
 
+func (s *Storage) ListUniqueSmartContractsByNetwork() ([]*smartcontract.SmartContract, error) {
+	// define smartcontracts response
+	smartcontracts := []*smartcontract.SmartContract{}
+
+	// get unique smartcontracts by network from db
+	/* @dev: It creates a sub table with a partition with only address and network fields.
+	 * This partition makes a counter for each row of smart contracts that has the same address
+	 * and network. Then from that partition we only get the first row, ensuring that we are not
+	 * getting any smart contract with this repeated info using the row number counter.
+	 */
+	scQuery := `SELECT id, name, network, node_url, address, last_tx_block_synced, status, error, created_at, updated_At FROM (
+					SELECT *, ROW_NUMBER() OVER (PARTITION BY address, network) AS rn
+					FROM smartcontracts
+				) AS sq
+			WHERE sq.rn = 1`
```

Comment:
@nicobarra why you use `sq.rn = 1` ?

Reply:
Because the table sq contains a rn (row number) column that indicates how often the same row is repeated. By selecting only the number 1, I ensure that I will get only one contract address per network.